.github/workflows/build-engine.yml

```
@@ -662,3 +662,10 @@ jobs:
     with:
       version: ${{ needs.config.outputs.version }}
       repo_url: ${{ format('{0}/{1}', github.server_url, github.repository) }} # このリポジトリのURL
+
+  run-build-engine-container-workflow:
+    if: needs.config.outputs.version != ''
+    needs: [config, run-release-test-workflow]
+    uses: ./.github/workflows/build-engine-container.yml
+    with:
+      version: ${{ needs.config.outputs.version }}
```

> この変更で、latestがpushされなくなる

というのが一瞬わからなかったのですが、確かにここは起動しなさそうですね。以下のようにすれば従来通りにはなると思います。というより、`docker build`までだけでもCIとして回し続けた方がよいのではと思っています。

```suggestion
    needs: [config, run-release-test-workflow]
    uses: ./.github/workflows/build-engine-container.yml
    with:
      version: ${{ needs.config.outputs.version_or_latest }}
```

---

> `docker build`までだけでも

について補足すると、 #1662 の`skip_dockerhub`に加えて`skip_ghcr`を追加し、両方とも`false`で`workflow_call`する感じをイメージしています。

---

あ、たしかにですね！！
CIとしてmasterにpushされるたびに`build-container.yml`のビルドまでは動かしておきたい気持ちはあります！！

ここを`version: ${{ needs.config.outputs.version_or_latest }}`にすると`latest`タグの中身が置き換わってしまいますねぇ。。。
たぶん、masterにpushされるたびに最新安定版のdocker imageがビルドされてpushされる形になる。

このあたり入り組んでいて、どうすべきかかなり悩みますね。。。。。 😇 
とーーーーーりあえず、CI目的なら`build-container.yml`側に`on: push`書いてmasterブランチ更新のたびに動くのが良さそう。
なのでここは`if: needs.config.outputs.version != ''`のときのみ動く形にするのが良さそう･･･？

代わりに`build-container.yml`側に`masterにpushされるたびにCIが回るようにする`みたいなTODOかFIXMEコメント書くと良いかも。。

---

> [追記] devとかedgeとかでmasterブランチの最新版を出すのも十分にありだと思います。

masterブランチの最新版docker imageビルドは出せれば出したい気持ちもあります。
現状の構成だとたぶん結構ややこしい実装になりそうなんですよねー。。。。
一応思いついてるの書くと、例えばこんな感じとかになりそう：

* `build-container.yml`の引数に`最新版としてpushする`フラグを追加
* フラグがONなら`generate_docker_image_names.py`に渡す`version`に`dev`なり何なりのタグを渡す
* 同様に`build-engine.yml`にもフラグを追加
* `build-latest-dev.yml`から`build-engine.yml`をdispatchするときにフラグをtrueにする

（ちょっと関係ないけどメモ）
`build-latest-dev.yml`が作るreleaseの名前に次のマイナーバージョン名を含めなければ、↑の４工程全部いらなくて`build-latest-dev.yml`のdispatchで`version`に`dev`なり何なりを渡せば解決しそう。
なんでマイナーバージョン名含めてるのか追ってみたけど、わからなかった。。。
[ここ](https://github.com/VOICEVOX/voicevox/pull/1484)が初出･･･？

---

> 代わりに`build-container.yml`側に`masterにpushされるたびにCIが回るようにする`みたいなTODOかFIXMEコメント書くと良いかも。。

とりあえずそうしました。
[`ad362bc` (#1669)](https://github.com/VOICEVOX/voicevox_engine/pull/1669/commits/ad362bc909dfc7f825f383828477f90c5ac8c4ce)

---

ありがとうございます！！

これちょっとissue作った方が良いかもですね！　もちろんマージ後に。　（忘れそう）